### PR TITLE
Backport lazy require webassemblyjs using modules to webpack-4

### LINKED ISSUE #9859: Backport to webpack 4: lazy require webassemblyjs using modules
https://github.com/webpack/webpack/pull/9857 needs to be backported to webpack 4.

Send a PR.

cc @sokra

### PR DESCRIPTION
**Pull request by jeffin143**
fixes #9859 

**What kind of change does this PR introduce?**
chore

**Did you add tests for your changes?**
Existing test

**Does this PR introduce a breaking change?**
No

**What needs to be documented once your changes are merged?**
Nothing

### CODE REVIEW

**Comment by sokra** [comment]
This PR is targeting the wrong branch. It should target the `webpack-4` branch

**Comment by sokra** [comment]
cherry-pick the commit onto the webpack-4 branch. Then force-push to this PR branch.

**Comment by jeffin143** [comment]
> cherry-pick the commit onto the webpack-4 branch. Then force-push to this PR branch.

Can u take a look now,

I tried to 
git fetch --all

but that doesn't fetch webpack-4 branch for some reasons

So i just cherry picket the commit to patch branch and then force pushed it 

Please let me know if that doesn't work

**Comment by sokra** [comment]
There should only be a single commit in the list above. Reset your branch to the `webpack-4` first. `git reset --hard webpack-4`.

File: lib/wasm-async/AsyncWebAssemblyModulesPlugin.js [hunk 1]
```diff
@@ -0,0 +1,187 @@
+/*
```
Comment (sokra) [inline lib/wasm-async/AsyncWebAssemblyModulesPlugin.js]: This file should not be here

File: lib/wasm/WebAssemblyModulesPlugin.js [hunk 2]
```diff
@@ -5,15 +5,17 @@
 "use strict";
 
 const Generator = require("../Generator");
-const WebAssemblyParser = require("./WebAssemblyParser");
-const WebAssemblyGenerator = require("./WebAssemblyGenerator");
-const WebAssemblyJavascriptGenerator = require("./WebAssemblyJavascriptGenerator");
-const WebAssemblyImportDependency = require("../dependencies/WebAssemblyImportDependency");
 const WebAssemblyExportImportedDependency = require("../dependencies/WebAssemblyExportImportedDependency");
+const WebAssemblyImportDependency = require("../dependencies/WebAssemblyImportDependency");
+const { compareModulesByIdentifier } = require("../util/comparators");
```
Comment (sokra) [inline lib/wasm/WebAssemblyModulesPlugin.js]:
```suggestion
```

File: lib/wasm-async/AsyncWebAssemblyModulesPlugin.js (hunk 1 above)
Comment (jeffin143) [reply lib/wasm-async/AsyncWebAssemblyModulesPlugin.js]: Removed :)

File: lib/wasm/WebAssemblyModulesPlugin.js (hunk 2 above)
Comment (jeffin143) [reply lib/wasm/WebAssemblyModulesPlugin.js]: Thanks for point out, Made apt changes

**Comment by sokra** [comment]
Thanks

**Pull request merged**